src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -105,7 +110,10 @@ protected void runTool(final JavaSparkContext ctx) {
         }
 
         final JavaRDD<GATKRead> initialReads = getReads();
-        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder());
+        if (parallelism == 0) {
+            parallelism = ReadsSparkSource.getRecommendedParallelism(ctx, getReadSourceName(), initialReads.partitions().size());
+        }
+        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder(), parallelism);
```

Comment:
Is `parallelism` really "the output parallelism" in this case, as the docs state, if it's also being used within `MarkDuplicatesSpark`? Should we perhaps separate the concepts of "output parallelism" and "processing/shuffle parallelism"?